region/tourian/main/Tourian Escape Room 4.json

```diff
@@ -139,11 +139,24 @@
           "HiJump",
           "SpaceJump",
           "canWalljump",
-          "h_canIBJ"
+          "h_canIBJ",
+          "canSpringBallJumpMidAir"
         ]}
       ],
       "note": "The pirates are free to kill, although they take some time."
     },
+    {
+      "link": [1, 3],
+      "name": "Shinespark",
+      "entranceCondition": {
+        "comeInShinecharged": {
+          "framesRequired": 15
```

Comment:
50 frames if just running in.  But entranceConditions have an implicit `canPrepareForNextRoom`?  Or at least you are aware of the next room since you know to try this strat.
Still should have `canShinechargeMovement` and `canMidairShinespark`

Reply:
Ah yeah, makes sense for this strat to have a `canPrepareForNextRoom` requirement since if you fall down without getting the spark off in time then you may be unable to get back to the door to retry. And I always forget about those other requirements.

Spin-jumping through the door seems like the reasonable thing to assume, though if the previous room is water (e.g. from doing a water shinecharge involving two rooms to the left) then the shinespark frames need to be a bit higher since the spin jump in will be low. Adjusted this and also updated the docs for `comeInShinecharged`. An alternative is we could use `comeInShinechargedJumping` for this and add a `physics` field to it, which could be "any" in this case.